File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    answer = 0
    sc=[]
    for i in scoville:
        heapq.heappush(sc,i)
    
    while len(sc)>=2:
        a=heapq.heappop(sc)
        if a>=K:
            return 0

        b=heapq.heappop(sc)
        if b==0:
            return -1
        heapq.heappush(sc,a+b*2)
        answer+=1
        temp=heapq.heappop(sc)
        if temp>=K:
            return answer
        heapq.heappush(sc,temp)
        
    return -1

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        for (int i=0;i<scoville.length;i++){
            pq.add(scoville[i]);
        }
        
        while(!pq.isEmpty()){
            int m1 = pq.poll();
            if (m1>=K){
                return answer;
            }
            if(!pq.isEmpty()){
                int m2 = pq.poll();
                pq.add(m1+m2*2);
                answer+=1;
            }
            else{
                return -1;
            }
        }
        return answer;
    
    }
}